util/adminApi/auth.ts

```diff
@@ -42,11 +57,52 @@ export async function authenticate(
     };
   }
 
+  // Only rate limit when an ip has been provided
+  if (ip) {
+    let countingStartDate = new Date();
+    // Subtract one hour
+    countingStartDate = new Date(countingStartDate.getTime() - 60 * 60 * 1000);
+
+    // Count requests for ip (for rate limiting)
+    const requestCount = await prisma.loggedAdminApiRequests.count({
+      where: {
+        ip: ip,
+        createdAt: {
+          gte: countingStartDate,
+        },
+      },
+    });
+
+    // Limit requests for ips within the last hour
+    if (requestCount >= adminApiConfig.requestLimit) {
+      logger.error(
+        `Ip(=${ip}) had ${requestCount} requests within the last hour and will thus not be processed.`
+      );
+
+      if (ip) {
+        logAdminApiRequest(authToken, false, ip);
```

Comment:
Missing await leads to floating promise


```suggestion
        await logAdminApiRequest(authToken, false, ip);
```